### PR TITLE
Add support for mixins to Babel 6.x

### PR DESCRIPTION
**Pull request by samwgoldman**
If this looks good I'll backport to Babel 5.x. In particular I'm unsure if it's kosher to reuse `InterfaceExtends` as the type of `mixins` nodes. Should I create a `ClassMixins` type instead?

cc @sebmck @jeffmo

### CODE REVIEW

**Comment by samwgoldman** [comment]
I forgot to add the necessary generator behaviors. Fixing now...

**Comment by samwgoldman** [comment]
OK, GTG.

**Comment by thejameskyle** [comment]
All these changes to flow stuff need updates in:

- https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/flow.js
- babel-plugin-transform-flow-comments
- babel-plugin-transform-flow-strip-types

File: packages/babel-generator/src/generators/flow.js [hunk 1]
```diff
@@ -104,6 +104,10 @@ export function _interfaceish(node: Object) {
     this.push(" extends ");
     this.printJoin(node.extends, node, { separator: ", " });
   }
+  if (node.mixins.length) {
```
Comment (sebmck) [inline packages/babel-generator/src/generators/flow.js]: We'll need a `node.mixins` existence check here too.

**Comment by samwgoldman** [comment]
@thejameskyle I think I addressed your comment, but can you confirm?

**Comment by sebmck** [comment]
There's still a missing `mixins` visitor key that will need to be added [here](https://github.com/babel/babel/blob/master/packages/babel-types/src/definitions/es2015.js#L66).

**Pull request merged**